src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLogger.cs

```diff
@@ -31,186 +26,41 @@ internal ConsoleLogger(string name, ConsoleLoggerProcessor loggerProcessor)
 
         internal ConsoleLoggerOptions Options { get; set; }
 
+        [ThreadStatic]
+        internal static StringWriter _stringWriter;
+
         public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
         {
             if (!IsEnabled(logLevel))
             {
                 return;
             }
-
             if (formatter == null)
             {
                 throw new ArgumentNullException(nameof(formatter));
             }
-
-            string message = formatter(state, exception);
-
-            if (!string.IsNullOrEmpty(message) || exception != null)
-            {
-                WriteMessage(logLevel, _name, eventId.Id, message, exception);
-            }
-        }
-
-        public virtual void WriteMessage(LogLevel logLevel, string logName, int eventId, string message, Exception exception)
-        {
-            ConsoleLoggerFormat format = Options.Format;
-            Debug.Assert(format >= ConsoleLoggerFormat.Default && format <= ConsoleLoggerFormat.Systemd);
-
-            StringBuilder logBuilder = _logBuilder;
-            _logBuilder = null;
-
-            if (logBuilder == null)
-            {
-                logBuilder = new StringBuilder();
-            }
-
-            LogMessageEntry entry;
-            if (format == ConsoleLoggerFormat.Default)
-            {
-                entry = CreateDefaultLogMessage(logBuilder, logLevel, logName, eventId, message, exception);
-            }
-            else if (format == ConsoleLoggerFormat.Systemd)
-            {
-                entry = CreateSystemdLogMessage(logBuilder, logLevel, logName, eventId, message, exception);
-            }
-            else
-            {
-                entry = default;
-            }
-            _queueProcessor.EnqueueMessage(entry);
-
-            logBuilder.Clear();
-            if (logBuilder.Capacity > 1024)
-            {
-                logBuilder.Capacity = 1024;
-            }
-            _logBuilder = logBuilder;
-        }
-
-        private LogMessageEntry CreateDefaultLogMessage(StringBuilder logBuilder, LogLevel logLevel, string logName, int eventId, string message, Exception exception)
-        {
-            // Example:
-            // info: ConsoleApp.Program[10]
-            //       Request received
-
-            ConsoleColors logLevelColors = GetLogLevelConsoleColors(logLevel);
-            string logLevelString = GetLogLevelString(logLevel);
-            // category and event id
-            logBuilder.Append(LoglevelPadding);
-            logBuilder.Append(logName);
-            logBuilder.Append('[');
-            logBuilder.Append(eventId);
-            logBuilder.AppendLine("]");
-
-            // scope information
-            GetScopeInformation(logBuilder, multiLine: true);
-
-            if (!string.IsNullOrEmpty(message))
+            if (exception == null && formatter(state, exception) == null)
```

Comment:
I don't think we want to call the `formatter` here and throw away the result. That will probably cause more string allocations than we want.